.rhdh/scripts/install-registry-proxy-images.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+#
+#  Copyright (c) 2024 Red Hat, Inc.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Ensure you have skopeo, podman, jq, yq, opm, and oc installed and configured.
+# This script assumes access to a connected environment for pulling images from quay.io.
+
+set -e
+
+# Variables
+REMOTE_REGISTRY="quay.io/rhdh"
+LOCAL_REGISTRY="image-registry.openshift-image-registry.svc:5000"
+NAMESPACE="openshift-marketplace"
+CATALOG_NAME="my-catalog-source"
+IIB_IMAGE="quay.io/repository/rhdh/iib"
+WORKDIR="/tmp/iib_extract"
+
+mkdir -p "$WORKDIR"
+
+# Step 1: Pull the IIB image and extract catalog information
+echo "Pulling IIB image and extracting catalog information..."
+skopeo copy --all "docker://$IIB_IMAGE" "oci:$WORKDIR/iib"
```

Comment:
Have you tested this?? The value you're using for the IIB is invalid.

```
IIB_IMAGE="quay.io/repository/rhdh/iib"
WORKDIR="/tmp/iib_extract"
mkdir -p "$WORKDIR"
skopeo copy --all "docker://$IIB_IMAGE" "oci:$WORKDIR/iib"

FATA[0000] initializing source docker://quay.io/repository/rhdh/iib:latest: reading manifest latest in quay.io/repository/rhdh/iib: unauthorized: access to the requested resource is not authorized 
```

Reply:
I have tested with the latest IIB image `quay.io/rhdh/iib:latest-v4.14-x86_64` this problem has been fixed, the execution output is now successful:

```bash
Processing manifest: sha256:71b67d678fb1b6c956ebaff5eaedfe5c60fdcfe150efac5b7050e3c43097e5f8
Analyzing JSON file: /tmp/iib_extract/iib/index.json
Converting image references to Quay.io equivalents...
Pulling and pushing images...
Creating CatalogSource...
catalogsource.operators.coreos.com/my-catalog-source created
CatalogSource my-catalog-source has been created and is using images from the local registry.
```